.pre-commit-config.yaml

```diff
@@ -0,0 +1,22 @@
+ci:
+  autofix_prs: false
+  autoupdate_schedule: 'quarterly'
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+    - id: check-added-large-files
+      args: ['--maxkb=1000']
+    - id: detect-aws-credentials
+      args: [--allow-missing-credentials]
+    - id: detect-private-key
+    - id: mixed-line-ending
+    - id: trailing-whitespace
+    - id: no-commit-to-branch
+      args: [--branch, main, --branch, dev]
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: 'v0.3.4'
+    hooks:
+    - id: ruff
+      args: [--fix]
+    - id: ruff-format
```

Comment:
Might be worth it to add mypy, or poetry-lock and poetry-check. I noticed the imap_processing pre-commit has these.